test/system/src/test/resources/pravega_withAuth.properties

```diff
@@ -31,9 +31,9 @@ controller.security.auth.delegationToken.signingKey.basis=secret
 autoscale.controller.connect.security.auth.enable=true
 autoscale.security.auth.token.signingKey.basis=secret
 #4GB of cache.
-pravegaservice.cache.size.max=4294967296
-#4GB of cache + a buffer for whatever else Netty might need it for.
-io.netty.maxDirectMemory=5368709120
+pravegaservice.cache.size.max=3094967296
```

Comment:
The cache is using Netty DM, so when you configure DM, it needs to be inclusive of the cache (plus the extra buffer you mentioned).

Reply:
ok..since cahce is using netty DM, in the example above if we set Netty DM to 1G, would that limit Cache's direct memory use to 1G, since Cache internally uses Netty ByteBuf? In that case do we need to set Netty DM?

Reply:
We need to set `XX:MaxDirectMemory` which is a Java-wide setting; Netty uses that value in case you don't specify something explicit for its own parameter (note the reverse is not true: just setting the Netty one will have no effect on the JVM one).

I recommend we set `XX:MaxDirectMemory` to whatever we need to - that will pool together all Netty and non-Netty allocated memory which is what we want. 

Note that the Cache size is set separately and it doesn't look at any of these settings. So if we mess up the configuration (i.e., cache size > DM size), the Cache will continue to request extra memory until the JVM will refuse to do so. We do not want to get into a situation like this.

Reply:
Thanks. Have put a precondition check for DM > cache